refactor: revamp snippet page

CMI-James opened the pull request:
<!-- Do not delete this PR template. Just edit it to include the required information -->

# Description

<!-- If your PR fixes an open issue, use `Closes #999` to link your PR with the issue. #999 stands for the issue number you are fixing -->

<!-- Github Issue Example: Closes #31 -->

**Closes #96**

# Changes proposed

## What were you told to do?

Make the /snippet page more visually appealing and match the color theme of the home page

## What did you do?

- Added animated background effects with teal/blue gradient orbs matching home page
- Implemented hero section with gradient text and stats cards
- Created modern search/filter section with backdrop blur and consistent styling
- Enhanced snippet cards with hover animations, language icons, and gradient backgrounds
- Added loading states, empty states, and call-to-action sections
- Applied consistent rounded-2xl borders and teal/blue color scheme throughout
- Integrated Framer Motion animations for smooth interactions
- Created responsive grid/list view toggle functionality

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

<!-- Mark all the applicable boxes. To mark the box as done follow the following conventions -->

<!--
[x] - Correct; marked as done
[X] - Correct; marked as done
[ ] - Correct; marked as **not** done

[] - Not Correct; syntax error
[ x] - Not Correct; space between the brackets
-->

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos
https://www.loom.com/share/aee7d144f2d1481fa0729c29e10f3343
<!-- If the changes are static page changes or UI changes add screenshots -->
<!-- If the changes involve implementing a functionality or working with apis, include a video
detailing how to implement the functionality and the request to the api and responses from the api endpoint-->
<!-- Add all the screenshots/videos which support your changes i.e before your change and after your change -->

SudiptaPaul-31 commented:
@CMI-James I think there's some issue with the loom video share some other video or share screenshots

SudiptaPaul-31 commented:
![image](https://github.com/user-attachments/assets/3f758fb6-d031-46a2-9964-64ae802d91b1)
@CMI-James What's the point of this kind of changes?

SudiptaPaul-31 commented:
![image](https://github.com/user-attachments/assets/08bddad0-7990-4a7b-b846-71ad83cc0055)
@CMI-James The task was to revamp this page ok

CMI-James commented:
@SudiptaPaul-31 Here’s a working demo video:
https://www.loom.com/share/1b16a5cfb46b4e8ebb01fc8060c04c70

The reason for the multiple file changes is that I updated the UI for the following routes:

 `/snippet` ,`/snippet/[id]` ,`/snippet/new`

Some components also needed minor adjustments to maintain a consistent and clean look across these pages that I updated.

However, if it feels like I went overboard with the changes, I’m happy to revert and focus solely on improving `/snippet` using the existing component structure.

Apologies for any inconvenience caused. let me know how you prefer.

SudiptaPaul-31 reviewed (commented):
Good work
LGTM
